### PR TITLE
Add more setup info to `check-pr` docs

### PR DESCRIPTION
**Pull request by ranweiler**
- Describe what permissions to grant the GitHub API `PAT`
- Document required env vars
- Recommend the use of `virtualenv` and `direnv`
- Note that `-h` can be used for more usage info

### CODE REVIEW

File: src/utils/check-pr/README.md [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
 
-* Python packages (See `requirements.txt`)
+* Python >= 3.7, and dependencies (see `requirements.txt`)
 * [az-cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
 * [azcopy](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10)
 * [azure-functions-core-tools](https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v3%2Clinux%2Ccsharp%2Cportal%2Cbash%2Ckeda#install-the-azure-functions-core-tools)
+
+## Setup
+
+1. In GitHub, generate a personal access token (PAT) with the `public_repo` scope.
```
Comment (ranweiler) [inline src/utils/check-pr/README.md:12]: Some `check-pr` functionality may require more scopes (unsure). This is definitely enough to run integration tests on a PR.

Comment (bmc-msft) [reply src/utils/check-pr/README.md:12]: Note, the only scope I use is `workflow`

Comment (ranweiler) [reply src/utils/check-pr/README.md:12]: This makes me wonder if `workflow` implies `public_repo`.

The API docs state that Artifacts access requires the [`[public]_repo` scope](https://docs.github.com/en/rest/reference/actions#artifacts).

**Review by chkeita — APPROVED** [review]

File: src/utils/check-pr/README.md (hunk 1 above)
Comment (ranweiler) [reply src/utils/check-pr/README.md:12]: Confirmed: when creating a GitHub PAT, if you check `workflow`, it auto-checks the entire `repo` scope _category_ (and disables un-checking), so it is much more permissive than `public_repo`.

**Pull request merged**